app/frontend/errors/errorMessages.ts

```diff
@@ -180,12 +142,68 @@ const translations = {
+  [knownExternalErrors.PoolRegNoTtl]: () =>
+    'TTL parameter is missing in the transaction. It is explicitly required even for the Allegra era.',
+  [knownExternalErrors.PoolRegNotTheOwner]: () =>
+    'This HW device is not an owner of the pool stated in registration certificate.',
+  [knownExternalErrors.PoolRegInvalidFileFormat]: () =>
+    'Specified file is not a cli-format pool registration certificate transaction.',
+  [knownExternalErrors.PoolRegWithdrawalDetected]: () =>
+    'The transaction must not include withdrawals.',
+  [knownExternalErrors.PoolRegInvalidType]: () =>
+    'The certificate in transaction is not a pool registration.',
+  [knownExternalErrors.PoolRegInvalidNumCerts]: () =>
+    'The transaction must include exactly one certificate, being the pool registration.',
+  [knownExternalErrors.TrezorRejected]: () =>
+    'TrezorRejected: Operation rejected by the Trezor hardware wallet.',
+  [knownExternalErrors.DaedalusMnemonic]: () => '',
+
+  [knownExternalErrors.TransportOpenUserCancelled]: ({message}) => {
+    const errors = {
+      'navigator.usb is undefined':
+        'Your browser does not support WebUSB, use e.g. Google Chrome instead.',
+    }
+
+    return `TransportCanceledByUser: ${message}. ${errors[message] || ''}`
+  },
+  [knownExternalErrors.TransportError]: ({message}) =>
+    `TransportError: ${message}.If you are using a hardware wallet, please make sure you are using the latest version of the Cardano application.`,
+  [knownExternalErrors.TransportStatusError]: ({message}) => {
+    const errors = {
+      'Failed to sign with Ledger device: U2F DEVICE_INELIGIBLE': ledgerTroubleshootingSuggestion,
+    }
+    return `TransportError: ${message}. ${errors[message] || updateHwWalletAppSuggestion}`
+  },
+  [knownExternalErrors.TransportInterfaceNotAvailable]: ({message}) => {
+    const errors = {
+      'Unable to claim interface.':
+        'Please make sure that no other web page/app is interacting with your Ledger device at the same time.',
+    }
+    return `TransportInterfaceNotAvailable: ${message} ${errors[message] || ''}`
+  },
+  [knownExternalErrors.DisconnectedDeviceDuringOperation]: () =>
+    `DisconnectedDeviceDuringOperation: ${ledgerTroubleshootingSuggestion}`,
+  [knownExternalErrors.TransportWebUSBGestureRequired]: () =>
+    `TransportWebUSBGestureRequired: ${ledgerTroubleshootingSuggestion}`,
+  [knownExternalErrors.NotFoundError]: () => `NotFoundError: ${ledgerTroubleshootingSuggestion}`,
+  [knownExternalErrors.AbortError]: () => `NotFoundError: ${ledgerTroubleshootingSuggestion}`,
+  [knownExternalErrors.SecurityError]: () => `Access denied: ${ledgerTroubleshootingSuggestion}`,
+  [knownExternalErrors.RedundantStakePool]: () => 'This stake pool is already chosen.',
+  [knownExternalErrors.DelegationFeeError]: () => 'Unsuccessful delegation fee calculation.',
+}
+
+function getErrorMessage(code: InternalErrorReason | string, params = {}) {
```

Comment:
Related to part of @refi93 comment:
```
provided the getTranslation API is refactored to receive the error class itself, which should be straightforward as we are currently just extracting the code from errors when invoking getTranslation()
```
So this is planned to be reworked in subsequent PR and are currently all enum keys in those enums distinct so that they do not clash for some values? If not addressed in here and there is no clash, please at least add a comment that it should be done. Thanks

Also I do not see it suggested in here https://github.com/vacuumlabs/adalite/issues/890? :thinking:

Reply:
I made sure they do not clash, I added a todo comment and also updated the issue description. :+1: Previously it was not specifically mentioned only because the `getErrorMessage` relates to `how we set errors to state`.